daft/execution/physical_plan.py

```diff
@@ -738,7 +741,7 @@ def materialize(
         # Check if any inputs finished executing.
         while len(materializations) > 0 and materializations[0].done():
             done_task = materializations.popleft()
-            yield done_task.partition()
+            yield done_task._result
```

Comment:
we may want to make `result` a method that also verifies that it has been set and is not None